claripy/simplifications.py

```diff
@@ -195,6 +197,22 @@ def eq_simplifier(a, b):
         if a.op == 'BVV' and b.op != 'BVV':
             return b == a
 
+        # expr - c == 0    ->   expr == c
+        if a.op == "__sub__" and a.args[1].op == "BVV" and b.op == "BVV":
+            return a.args[0] == a.args[1] + b
+
+        # 1 ^ expr == 0     ->   expr == 1
+        if a.op == '__xor__' and isinstance(b.args[0], int) and b.args[0] == 0:
+            if a.args[0].op == 'If' and a.args[1].args[0] == 1:
+                return a.args[0] == 1
```

Comment:
These 3 lines read to me as transform:
`( If[v1, v2, v3] ^ op1[1, ...] ^ ...) == b.op[0, ...]` into `If[v1, v2, v3] == 1` 
Where `op1` and `b.op` are unknown ops and where `...` indicates that there may or may not be more arguments, it is not restricted. Something seems wrong here.

Reply:
The following 2 lines have a similar interpretation